### ISSUE TITLE
Error code 252 when analysis ends with different paths

### ISSUE THREAD

**Issue by Fischer96** [sylvainhalle/textidote#112]
I'm using Textidote in a Github Action after building my latex pdf file. Everything is running in docker containers and Textidote is installed as a debian package. The following is giving me error code 248 or 252 depending on the path to the .tex file. When the file is in the same folder as my output file which is generated, then the app throws code 248. Otherwise 252. The error message always looks like this: 

`##[error]Process completed with exit code 248.`

And thats all.

A simple fix is ignoring the error code like this:

`textidote --output html --check de thesis.tex > report.html || true`

Even though i ignore the error the application is doing fine, my html file looks good and no problems at all. Would be nice to know why this is happening.

**Comment by sylvainhalle** [comment]
Thanks for your feedback. I am not sure I completely understand the issue. Is it TeXtidote that ends with code 248/252? Could you provide a minimum working example that reproduces the problem?

**Comment by Fischer96** [comment]
I'm running an Github Action like this:

```yaml
name: SpellCheck

on:
  push:
    branches:
      - master

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1
    - name: Spell and Grammar Check
      run: |
        wget -q https://github.com/sylvainhalle/textidote/releases/latest/download/textidote_0.8.1_all.deb
        sudo apt-get install ./textidote_0.8.1_all.deb
        cd thesis/chapters/thesis
        mkdir -p reports
        textidote --output html --check de chapter01.tex > reports/chapter01.html || true
        textidote --output html --check de chapter02.tex > reports/chapter02.html || true
        textidote --output html --check de appendix01.tex > reports/appendix01.html || true
    - name: Upload Language Report as Artifact
      uses: actions/upload-artifact@master
      with:
        name: Report
        path: thesis/chapters/thesis/reports
```

Lets imagine the file chapter01.tex is in test/test123/chapter01.tex. The tool would then throw error 252. In the current situation above its throwing 248. Both of these errors can be avoided with `|| true` at the end. Those errors are only thrown when using Github Actions. On a local machine nothing happens. But when suppressing the error with `|| true` everything is still working fine. So my guess is that Textidote or any other tool or library it may use, answers with a wrong return code even though everything is fine.

I hope you understood what i mean!

**Comment by sylvainhalle** [comment]
The "problem" lies here: https://github.com/sylvainhalle/textidote/blob/294a8e4ea64290a1c862816865c637d5f3d0ed16/Source/Core/src/ca/uqac/lif/textidote/Main.java#L660

The return code is the number of warnings/errors found in a document, which is most likely non-zero. It seems that the GitHub action interprets a non-zero exit code as a failure and stops.

**Comment by Fischer96** [comment]
That seems plausible, so i suppose it's up to GitHub to not interpret values like this as error codes

**Comment by sylvainhalle** [comment]
That said, I don't think this exit code "feature" is very useful in TeXtidote. We could change the program to return 0 except in case of a real failure.

**Comment by Fischer96** [comment]
Should be an easy fix i guess. I will look into it if you want

**Comment by Fischer96** [comment]
Merged fix #113
